project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.utilities;
+
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.sps.data.User;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DatabaseWrapper {
+  private String instanceID;
+  private String databaseID;
+  private static final String USER_TABLE = "Users";
+
+  public DatabaseWrapper(String instanceID, String databaseID) {
+    this.instanceID = instanceID;
+    this.databaseID = databaseID;
+  }
+
+  /**
+   * Given a user, insert a row with all available fields into the DB
+   *
+   * @param user the user to be inserted; user's ID field should not exist in DB
+   */
+  public void insertUser(User user) {
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceID, databaseID);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
+
+    List<Mutation> mutations = newMutationFromUser(user, false);
+    dbClient.write(mutations);
+    spanner.close();
+  }
+
+  /**
+   * Given a user, insert a row with all available fields into the DB
+   *
+   * @param user the user to be updated; user's ID field should already exist in DB
+   */
+  public void updateUser(User user) {
+    // Given a user, update its corresponding row's new fields in DB
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceID, databaseID);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
+
+    List<Mutation> mutations = newMutationFromUser(user, true);
+    dbClient.write(mutations);
+    spanner.close();
+  }
+
+  private static List<Mutation> newMutationFromUser(User user, boolean update) {
```

Comment:
boolean parameter is not recommended. Please check this doc and choose the solution you prefer:

https://docs.google.com/document/d/19djty1fSEpp5vCmhWJ7x4Dakt9NfBIwN6wWhNDbGhL0/edit#

Reply:
BTW, I think since it is just a private util function, it not needed to bother yourself to use 2 two uniquely named methods. Try one solution of the others.

Reply:
Two uniquely-named methods seems most appropriate here.